enable/trait_defs/ui/editor_with_component.py

```diff
@@ -0,0 +1,133 @@
+# (C) Copyright 2005-2022 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+from traits.api import Bool, Instance, Str, observe
+from traitsui.api import BasicEditorFactory, Editor as BaseEditor, toolkit_object
+
+from enable.component import Component
+from enable.label import Label
+from enable.window import Window
+from enable.enable_traits import font_trait
+
+
+Editor = toolkit_object("editor:Editor")
+if not issubclass(Editor, BaseEditor):
+    Editor = object
```

Comment:
I'm assuming that this is so that the import doesn't break if the toolkit is "null"; is that correct? It might be worth an explanatory comment.

Reply:
Yes, I believe so; this behaviour was copied from earlier code.  It is possible it may not be needed any more.

Reply:
Yep, still needed here; may be removable in #929 which re-works the trait which uses the editor.